lib/frp/src/core/node/id.rs

```diff
@@ -0,0 +1,49 @@
+//! This module defines FRP node identifiers. They are mainly used for debugging purposes.
+
+use crate::prelude::*;
+
+
+
+// =============
+// === HasId ===
+// =============
+
+/// Each FRP node is assigned with an unique ID. This is currently used mainly for debugging
+/// purposes.
+pub trait HasId {
```

Comment:
Few things:

1. Big traits are not composable. I did it in the past a lot (in Haskell) and I learned that it is always the bad way to go.
2. You should be happy now, I cleaned up a lot of trait aliases and there is now `AnyNode` trait which collects traits for nodes.
3. Having separate traits is super cool. For example here, your custom nodes do not need to implement `HasId` as it is automatically implement by the wrapper node! So although it is requirement on every node, it can be implemented by a "layer above node", while some of the traits can be redirected to the concrete node implementation. See the file `class.rs` and `prim.rs` to see examples of that.